### PR TITLE
[yeelight] Add more devices and include lib directly in source code

### PR DESCRIPTION
**Pull request by claell**
PR for adding new devices (#3665) and lib improvements (#3667)

Latest build from this PR is available at https://openhab.jfrog.io/openhab/libs-pullrequest-local/org/openhab/binding/org.openhab.binding.yeelight/2.4.0-SNAPSHOT/ for testing.

Signed-off-by: Claudius Ellsel <claudius.ellsel@live.de>

### CODE REVIEW

**Comment by claell** [comment]
@daw87 For the future: Commits need signing to be accepted in this repo, see https://www.openhab.org/docs/developer/contributing/contributing.html#sign-your-work
The handler should be ok, essentially they are doing the same, so no need to change them. Thus I set your changes back with my latest commit.

**Comment by Kaos1337** [comment]
Thank you, with this PR openhab can detect the white bulb v2.

**Comment by mohlek** [comment]
whats the problem with this pr? Can it be merged?

**Comment by kaikreuzer** [comment]
As long as it is marked as [WIP], it won't be merged.
So @claell, please let us know once it is ready for review!

**Comment by mohlek** [comment]
i'd love to see this upstream. I recently brought a few devices and it seems that this would solve a few problems and supports more devices.

**Comment by claell** [comment]
Sorry, I pretty much forgot about this PR. I agree that this should be merged soon.

Apparently there had been some problems which were mentioned in #3665 but the people mentioning them did not respond always and I am lacking the devices to try it myself, as they are probably device specific.

@mohlek recently did a PR to my repo adding further support, see https://github.com/claell/openhab2-addons/pull/3. I think after this is merged the WIP can be removed.

**Comment by claell** [comment]
@kaikreuzer Should be fine from my side now. The only problem are the reports of problems in #3665 which could not really get tracked down and are probably rather device specific. Also afaik it was not sure that the problems weren't there before (in the older version). So from my point of view this is an improvement and should be merged.

However there still might be some flaws in the binding, probably related to the yeelight library which is now integrated directly in the source with this PR.

**Comment by mohlek** [comment]
@claell I think you have to sign the merge commit.

**Comment by kaikreuzer** [comment]
> I think you have to sign the merge commit.

Should be ok as it is. But @claell, please NEVER do merge commits, always instead only do rebase on latest master of your branch.

**Comment by claell** [comment]
Alright, will try to do so if this happens again in the future. I guess this avoids having the PR commit to my repo and additionally my merge of the PR appearing in the commits for this PR (https://github.com/openhab/openhab2-addons/pull/3673/commits)?

File: addons/binding/org.openhab.binding.yeelight/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -20,7 +18,19 @@ Import-Package:
...
+ org.openhab.binding.yeelight.lib.device.connection,
+ org.openhab.binding.yeelight.lib.enums,
+ org.openhab.binding.yeelight.lib.listeners,
+ org.openhab.binding.yeelight.lib.services,
  org.slf4j
 Service-Component: OSGI-INF/*.xml
 Export-Package: org.openhab.binding.yeelight,
- org.openhab.binding.yeelight.handler
+ org.openhab.binding.yeelight.handler,
+ org.openhab.binding.yeelight.lib,
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.yeelight/META-INF/MANIFEST.MF]: As the yeelight lib packages haven't been exposed before, there is probably no need to do so now.
Could you rename the package to `org.openhab.binding.yeelight.internal.lib` and remove these exports?

File: addons/binding/org.openhab.binding.yeelight/build.properties [hunk 2]
```diff
@@ -4,7 +4,6 @@ bin.includes = META-INF/,\
                .,\
                OSGI-INF/,\
                ESH-INF/,\
-               lib/yeelight.jar,\
                about.html,\
                about_files/
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.yeelight/build.properties]: I'd assume the about_files can be removed and the notice about the external lib can be removed from about.html as well.

**Comment by mohlek** [comment]
Any progress?

**Pull request merged**